### DISCUSSION TITLE
Store files in a folder

### DISCUSSION THREAD

**Discussion by G-JT** [serverpod/serverpod#802] [Q&A] [answered]
Hello,

I want to set up Serverpod to replace the current backend of the [Dotspot](https://dotspot.fr) project. However, I have a doubt to clear up before setting it up.
Currently, it is possible to store files on S3 or in a database. My need is to store my files directly in a folder.

Do you think this is possible in the current version of Serverpod?

Thanks for your help

**Comment by vlidholt** [comment, marked as answer]
Yes, you can just use the regular `dart:io` package to store files locally. You obviously want to be a bit mindful on where you are hosting your server, so that you have access to the drive and that it isn't ephemeral.

I would suggest that you use the database cloud storage as a template for getting started. Create a class called `LocalFileStorage`. Just replace the places where it stores data to the database with writing to the disk. Should be very straight forward.
https://github.com/serverpod/serverpod/blob/main/packages/serverpod/lib/src/cloud_storage/database_cloud_storage.dart

**Reply by G-JT** [reply]
Thanks
